File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int n = nums.length;
        int element ;
        int excess;
        HashMap<Integer,Integer> mpp = new HashMap<>();

        for(int i=0;i<n; i++) {
            element = nums[i];
            excess = target - element;

            if(mpp.containsKey(excess) ) {
                return new int[] {mpp.get(excess),i};
            }
            mpp.put(element,i);
        }   
        return new int[] {};
    }
}

File: 0001-two-sum/0001-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        mpp = {}  

        for i, element in enumerate(nums):
            excess = target - element

            if excess in mpp:
                return [mpp[excess], i]

            mpp[element] = i

        return []
